src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/DataCollectionLauncher.cs

```diff
@@ -65,33 +58,18 @@ public void Initialize(Architecture architecture)
         public virtual int LaunchDataCollector(IDictionary<string, string> environmentVariables, IList<string> commandLineArguments)
         {
             var currentWorkingDirectory = Path.GetDirectoryName(typeof(DataCollectionLauncher).GetTypeInfo().Assembly.Location);
-            string dataCollectorProcessPath, processWorkingDirectory = null;
+            string dataCollectorProcessPath = null, processWorkingDirectory = null;
```

Comment:
This is handled in ProxyExecutionManagerWithDataCollection, the class responsible for initiating test execution when data collection is enabled.